ecommerce/extensions/payment/processors/paystack.py

```diff
@@ -0,0 +1,200 @@
+""" Paystack payment processor. """
+from __future__ import absolute_import, unicode_literals
+
+import logging
+
+from django.urls import reverse
+from django.utils.functional import cached_property
+from oscar.apps.payment.exceptions import GatewayError
+from oscar.core.loading import get_model
+
+from ecommerce.core.url_utils import get_ecommerce_url
+from ecommerce.extensions.payment.exceptions import RefundError
+from ecommerce.extensions.payment.processors import BaseClientSidePaymentProcessor, HandledProcessorResponse
+from ecommerce.qverse_features.paystack.client import PaystackClient
+from ecommerce.qverse_features.paystack.constants import (
+    CREATE_REFUND_CODE,
+    FETCH_REFUND_CODE,
+    INITIALIZE_TRANSACTION_CODE
+)
+
+logger = logging.getLogger(__name__)
+
+
+class Paystack(BaseClientSidePaymentProcessor):
+    NAME = 'paystack'
+
+    def __init__(self, site):
+        """
+        Constructs a new instance of the Paystack processor.
+
+        Raises:
+            KeyError: If no settings configured for this payment processor.
+        """
+        super(Paystack, self).__init__(site)
+        self.public_key = self.configuration['public_key']
+        self.secret_key = self.configuration['secret_key']
+
+    @cached_property
+    def paystack_client(self):
+        """
+        Returns paystack client instance with appropriate configuration
+        """
+        return PaystackClient(self.configuration['base_url'], self.secret_key)
+
+    @property
+    def cancel_url(self):
+        return get_ecommerce_url(self.configuration['cancel_checkout_path'])
+
+    def get_return_url(self):
+        redirect_url = reverse('paystack:execute')
+        ecommerce_base_url = get_ecommerce_url()
+        return '{}{}'.format(ecommerce_base_url, redirect_url)
+
+    def get_basket_amount(self, basket):
+        return str((basket.total_incl_tax * 100).to_integral_value())
+
+    def get_paystack_custom_fields_data(self, basket):
+        """
+        Return Paystack Custom fields.
+        """
+        product = basket.all_lines()[0].product
```

Comment:
Should we handle the `IndexOutOfRange` exception here?

Reply:
This should never happen. And if in any scenario we face it, then the server should be crashed.